crates/ink/ir/src/ir/chain_extension.rs

```diff
@@ -158,39 +221,100 @@ impl<'a> Iterator for ChainExtensionMethodInputs<'a> {
     }
 }
 
-/// The unique ID of an ink! chain extension method.
+/// The unique ID of an chain extension.
 ///
 /// # Note
 ///
-/// The ink! attribute `#[ink(extension = N: u32)]` for chain extension methods.
-///
-/// Has a `func_id` extension ID to identify the associated chain extension method.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+/// The ink! attribute `#[ink::chain_extension(extension = N: u16)]` for chain extension.
+#[derive(Default, Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ExtensionId {
-    index: u32,
+    index: u16,
 }
 
 impl ExtensionId {
-    /// Creates a new chain extension method ID from the given `u32`.
-    pub fn from_u32(index: u32) -> Self {
+    /// Creates a new chain extension ID from the given `u16`.
+    pub fn from_u16(index: u16) -> Self {
+        Self { index }
+    }
+
+    /// Returns the underlying raw `u16` index.
+    pub fn into_u16(self) -> u16 {
+        self.index
+    }
+}
+
+/// The unique ID of the method within the chain extension.
+///
+/// # Note
+///
+/// The ink! attribute `#[ink(function = N: u16)]` for chain extension methods.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct FunctionId {
+    index: u16,
+}
+
+impl FunctionId {
+    /// Creates a new chain extension function ID from the given `u16`.
+    pub fn from_u16(index: u16) -> Self {
         Self { index }
     }
 
+    /// Returns the underlying raw `u16` index.
+    pub fn into_u16(self) -> u16 {
+        self.index
+    }
+}
+
+/// The unique ID of a chain extension method across all chain extensions.
+///
+/// # Note
+///
+/// It is a combination of the [`ExtensionId`] and [`FunctionId`].
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct GlobalMethodId {
```

Comment:
In your example you access the method by specifying a chain extension instance first: `self.env().extension().rand.fetch_random([0; 32]`

Why do you need that if you can identify each method uniquely? Why can't you collect all chain extension functions into a single "list" to make them accessible together? 

e.g.
`self.env().extensions().fetch_random([0; 32]`

and then dispatch the call to the associated chain extension since you can derive this info from the function id

Reply:
If you have the method with the same name in each extension, it is impossible to identify which instance we need to call.

Reply:
It doesn't work if you want to use extensions defined in other crates.